File: Data Structures & Algorithms/two-integer-sum/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        const map = new Map();
        let difference;
        for(let i = 0; i < nums.length; i++){
            difference = target - nums[i];
            if(map.get(difference) !== undefined) return [i, map.get(difference)];
            if(map.get(nums[i]) === undefined)map.set(nums[i], i);
        }
    }
}

File: Data Structures & Algorithms/search-2d-matrix/submission-1.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        L = 0
        R = len(matrix) * len(matrix[0]) - 1
        while L <= R:
            mid = (L + R) // 2
            x = mid // len(matrix[0])
            y = mid % len(matrix[0])
            if target < matrix[x][y]:
                R = mid - 1
            elif target > matrix[x][y]:
                L = mid + 1
            else:
                return True
        return False
